addons/mail/models/mail_thread.py

```diff
@@ -2156,23 +2156,23 @@ def message_notify(self, partner_ids, body='', subject=False, **kwargs):
             author = self.env['res.partner'].sudo().browse(kw_author)
         else:
             author = self.env.user.partner_id
-        if not author.email:
-            raise exceptions.UserError(_("Unable to notify message, please configure the sender's email address."))
-        email_from = formataddr((author.name, author.email))
 
         msg_values = {
             'subject': subject,
             'body': body,
             'author_id': author.id,
-            'email_from': email_from,
             'message_type': 'notification',
             'partner_ids': partner_ids,
             'model': False,
             'subtype_id': self.env['ir.model.data'].xmlid_to_res_id('mail.mt_note'),
             'record_name': False,
-            'reply_to': self.env['mail.thread']._notify_get_reply_to(default=email_from, records=None)[False],
             'message_id': tools.generate_tracking_message_id('message-notify'),
         }
+        if author.email:
+            email_from = formataddr((author.name, author.email))
+            msg_values['email_from'] = email_from
+            msg_values['reply_to'] = self.env['mail.thread']._notify_get_reply_to(default=email_from, records=None)[False]
```

Comment:
It looks like _notify_get_reply_to will return the catchall most of the time here, not rally linked to the default. Now, we won't have a reply_to if we don't have an author.email. Is it intended?